BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h

```diff
@@ -0,0 +1,153 @@
+//
+//  BOXURLSessionCacheClient.h
+//  BoxContentSDK
+//
+//  Created by Thuy Nguyen on 3/3/17.
+//  Copyright © 2017 Box. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface BOXURLSessionTaskCachedInfo : NSObject
+
+@property (nonatomic, strong, readwrite) NSString *backgroundSessionId;
+@property (nonatomic, assign, readwrite) NSUInteger sessionTaskId;
+@property (nonatomic, strong, readwrite) NSString *destinationFilePath;
+@property (nonatomic, strong, readwrite) NSData *resumeData;
+@property (nonatomic, strong, readwrite) NSURLResponse *response;
+@property (nonatomic, strong, readwrite) NSData *responseData;
+@property (nonatomic, strong, readwrite) NSError *error;
+
+@end
+
+@protocol BOXURLSessionCacheClientDelegate <NSObject>
```

Comment:
IMO, unless we know/really think we're going to use it for something else, lets be specific at this point, it will make this stuff easier to consume/understand